web.py

```diff
@@ -139,5 +187,38 @@ def compile(emit, optimize, version, color, syntax):
         else:
             return {"result": split[1].decode()}
 
-os.chdir(sys.path[0])
-run(host='0.0.0.0', port=80, server='cherrypy')
+def main(args):
+    '''
+    -d: Debugging mode: turn on verbose debugging (and RUST_BACKTRACE=1)
+        Change the listening port to 8080 from the default 80.
+    -S: Skip playpen and direcly run rustc and the output binary.
+        To mitigate the risk, accepts only local connections by default.
```

Comment:
> since the behaviour under playpen has to be tested, since it differs from normal execution.

I see you point, but I still believe decoupling the functionality as a web app and jailing is important.  In fact I don't have a working playpen environment, not to mention jail directories `root-stable` etc whose structure is not documented.

> playpen: execvpe: No such file or directory

This must be because `compile.sh` etc are implicitly assumed to be installed as `root-stable/usr/local/bin/compile.sh` etc without any documentations.  We should proceed to improve portability rather than getting more and more tied to a particular deployment.

> Due to some interaction with multirust (I suspect it's because everything runs as a different user), I also can't get this to work without playpen:

That's odd, I don't see any problems with and without multirust-the-shellscript under the playpen-less mode.